src/cpucounters.cpp

```diff
@@ -1277,7 +1277,10 @@ bool PCM::discoverSystemTopology()
     }
 
     TopologyEntry *entries = new TopologyEntry[num_cores];
-    MSR[0]->buildTopology(num_cores, entries);
+    if (MSR[0]->buildTopology(num_cores, entries) != 0) {
+      std::cerr << "Unable to build CPU topology" << std::endl;
```

Comment:
could you deallocate the memory in the entries array to avoid the memory leak?

Reply:
Yeah, definitely! Sorry, I missed that after working on something else.  I will upload a new patch.

Reply:
Updated (it doesn't appear to me that it is necessary to allocate the array on the heap, and, as far as I could tell, nobody stores a pointer to it outside of this code), so I made it statically allocated.